### PR TITLE
Updates to shared docs   [ci skip]

### PR DESCRIPTION
**Pull request by mwichmann**
Mostly update markup to more "standard" markup conventions.

There are some substantive wording changes to `CacheDir` descriptions in `Environment.xml`, and to the `AddOption`/`Help` descriptions in `Main.xml` and `SConscript.xml` files. 

Doc-only change - affects both manpage and user guide.

Signed-off-by: Mats Wichmann <mats@linux.com>

## Contributor Checklist:

* [ ] I have created a new test or updated the unit tests to cover the new/changed functionality.
* [ ] I have updated `CHANGES.txt` (and read the `README.rst`)
* [X] I have updated the appropriate documentation

### CODE REVIEW

File: SCons/Environment.xml [hunk 1]
```diff
@@ -638,68 +633,61 @@ until after the Builder object is actually called.
...
-<function>env.CacheDir</function>()
-specified.
+that do not set up environment-specific
+caching by calling &f-env-CacheDir;.
 </para>
 
 <para>
-When a
-<function>CacheDir</function>()
+When cachin
```
Comment (bdbaddog) [inline SCons/Environment.xml]: ? typo?

File: SCons/Environment.xml [hunk 2]
```diff
@@ -1042,9 +1026,8 @@ A now-deprecated synonym for
 Specifies that all up-to-date decisions for
 targets built through this construction environment
 will be handled by the specified
-<varname>function</varname>.
-The
-<varname>function</varname>
+<parameter>function</parameter>.
+<parameter>function</parameter>
```
Comment (bdbaddog) [inline SCons/Environment.xml]: Shouldn't the second one here be "The <..."

Comment (mwichmann) [reply SCons/Environment.xml]: it's saying the `function` argument can also be a string. "The" isn't right, but I'll adjust wording.

File: SCons/Environment.xml [hunk 3]
```diff
@@ -638,68 +633,61 @@ until after the Builder object is actually called.
 </arguments>
 <summary>
 <para>
-Specifies that
+Direct
 &scons;
-will maintain a cache of derived files in
-<varname>cache_dir</varname>.
+to use a derived-file cache in
```
Comment (bdbaddog) [inline SCons/Environment.xml:638]: maintain is better here.  use doesn't imply contribute to?

**Pull request merged**